packages/react-devtools-extensions/src/main.js

```diff
@@ -237,52 +217,89 @@ function createPanelIfReactLoaded() {
         // never reaches the chrome.runtime.onMessage event listener.
         let fetchFileWithCaching = null;
         if (isChrome) {
-          // Fetching files from the extension won't make use of the network cache
-          // for resources that have already been loaded by the page.
-          // This helper function allows the extension to request files to be fetched
-          // by the content script (running in the page) to increase the likelihood of a cache hit.
-          fetchFileWithCaching = url => {
-            const event = cachedNetworkEvents.get(url);
-            if (event != null) {
-              // If this resource has already been cached locally,
-              // skip the network queue (which might not be a cache hit anyway)
-              // and just use the cached response.
-              return new Promise(resolve => {
-                event.getContent(content => resolve(content));
-              });
+          const fetchFromNetworkCache = (url, resolve, reject) => {
+            if (__DEBUG__) {
+              console.log('[main] fetchFromNetworkCache()', url);
             }
 
-            // If DevTools was opened after the page started loading,
-            // we may have missed some requests.
-            // So fall back to a fetch() and hope we get a cached response.
+            chrome.devtools.network.getHAR(harLog => {
+              for (let i = 0; i < harLog.entries.length; i++) {
+                const entry = harLog.entries[i];
+                if (url === entry.request.url) {
+                  entry.getContent(content => {
+                    if (content) {
+                      resolve(content);
+                    } else {
+                      if (__DEBUG__) {
+                        console.log(
+                          '[main] fetchFromNetworkCache() Invalid content returned by getContent():',
+                          content,
+                        );
+                      }
+
+                      // Edge case where getContent() returned null; fall back to fetch.
+                      fetchFromPage(url, resolve);
```

Comment:
Based on my testing, `content` is pretty much always `null` in Facebook _unless_ we pre-cache by calling `event.getContent()` from `network.onRequestFinished` – but this defeats the purpose.

Could this be related to the `Vary: 'Origin'` header?

Reply:
Trying this same code against a non-Facebook site (Code Sandbox) I can see that– even though the DevTools extension is open from before the page is loaded, it looks like none of the source file URLs are matched in the HAR log. That also seems unexpected?